File: plugins/webrtc/src/zygote.ts
import sdk, { PluginFork } from '@scrypted/sdk';
import worker_threads from 'worker_threads';

export type Zygote<T> = () => PluginFork<T>;

export function createZygote<T>(): Zygote<T> {
    if (!worker_threads.isMainThread)
        return;

    let zygote = sdk.fork<T>();
    function* next() {
        while (true) {
            const cur = zygote;
            zygote = sdk.fork<T>();
            yield cur;
        }
    }

    const gen = next();
    return () => gen.next().value as PluginFork<T>;
}

File: plugins/python-codecs/src/zygote.py
import scrypted_sdk
from typing import List


def createZygote():
    queue: List[scrypted_sdk.PluginFork] = []
    for i in range(0, 4):
        queue.append(scrypted_sdk.fork())

    def next():
        while True:
            cur = queue.pop(0)
            queue.append(scrypted_sdk.fork())
            yield cur

    gen = next()
    return lambda: gen.__next__()
